tools/clang/test/CodeGenSPIRV/cast.flat-conversion.literal-initializer.hlsl

```diff
@@ -47,4 +52,11 @@ void main() {
 // CHECK-NEXT:           {{%[0-9]+}} = OpCompositeConstruct %S [[a2_float]] [[a_float_0]] [[a2_double]] [[a]] [[a_int64]] [[a_uint64]]
   double a;
   S s1 = (S)(a);
+
+// TODO(6188): This is wrong because we lose most significant bits in the literal.
+// CHECK: [[lit:%[0-9]+]] = OpIAdd %int %int_0 %int_1
+// CHECK: [[longLit:%[0-9]+]] = OpSConvert %long [[lit]]
+// CHECK: [[t:%[0-9]+]] = OpCompositeConstruct %T [[lit]] [[longLit]]
+// CHECK: OpStore %t [[t]]
+  T t = (T)(0x100000000+1);
```

Comment:
Could you add lines to test suffixes?
```
T t = (T)(0x100000000ul+1);
T t = (T)(0x100000000ull+1);
```

There are behavior differences between C++ and DXIL compilation, and SPIR-V compilation and DXIL, so we might want to be careful to not diverge from DXIL (which already diverges from C++)